src/util/XournalType.cpp

```diff
@@ -24,8 +24,7 @@ using std::endl;
 
 static bool listInited = false;
 static const char* xojTypeList[XOURNAL_TYPE_LIST_LENGTH] = { 0 };
-
-GMutex* mutex = NULL;
+static GMutex mutex;
```

Comment:
I checked it, in my opinion it looks wrong. But yes, it's valid C99.
http://stackoverflow.com/questions/13251083/the-initialization-of-static-variables-in-c
https://developer.gnome.org/glib/stable/glib-Threads.html#GMutex

But we are compiling in C++ mode, I cannot find that C++ also have this roule.
To be sure, write please static GMutex mutex = {0};